Ansible Community Steering Committee Doc & Guidelines (#42)
ansible/community-docs: ansible_community_steering_committee.rst

************************************
Ansible Community Steering Committee
************************************

The Steering Committee’s role is to provide continuity, guidance, and

> suggestions to the Ansible, and to ensure delivery of the Ansible

Reviewer: ```suggestion
suggestions to the Ansible community, and to ensure delivery of the Ansible
```
wasn't sure if this was meant to be "the Ansible community" or just "Ansible"

Author: Good find! Thanks @briantist

to this `GitHub
discussion <https://github.com/ansible-collections/ansible-inclusion/discussions/new>`__
and requests regarding new policy/proposals should be submitted through
a new pull request to
`ansible-collections/overview <https://github.com/ansible-collections/overview>`__
GitHub repo.

Also, we earlier used a `Community Working Group Meeting
Agenda <https://github.com/ansible/community/issues/539>`__ to track
agenda topics that needed to be discussed in the weekly Ansible
Community IRC meetings. Now, to categorize each topic separately we’ve
created a new GitHub repo
“`community-topics <https://github.com/ansible-community/community-topics>`__”.
If you want to discuss an idea, improvements or submitted new
Policy/Proposals & New Collection Inclusion Requests then create a `new
issue <https://github.com/ansible-community/community-topics/issues/new>`__
in the `repo <https://github.com/ansible-community/community-topics>`__
as a topic, and it’ll be discussed publicly in the weekly IRC meeting.
After every meeting, the meeting minutes/summaries and meeting logs will
be posted to the `original
issue <https://github.com/ansible/community/issues/539>`__.

> If you cannot join the meeting or from another time zone then you can

Reviewer: ```suggestion
If you cannot join the meeting or you are in another time zone then you can
```

Reviewer 2: I think we could remove `or you are in another time zone` because this also mean `you cannot join the meeting` in that case

Author: Agreed!

************************************
Ansible Community Steering Committee
************************************

The Steering Committee’s role is to provide continuity, guidance, and
suggestions to the Ansible, and to ensure delivery of the Ansible
Community package. The Steering Committee should reflect the scope and
breadth of the Ansible Community.

Mission:
========

The Steering Committee helps decide the technical direction of the
Ansible Upstream project and is also responsible for avowal of new
proposals (feature idea, new policy in collection world), new collection
requests, empower contributors, maintainers & reviewers and other
technical aspects regarding inclusion and packaging.

Committee Member Responsibilities:
==================================

The members in the committee handle the below:

- Design policy and procedure for the collection world.
- Review things (collections fornow) for compliance with the policies. 
- Help create our deliverables (community.general, community.network, the ansible-tarball, docs website).
- Ballot on approval changes to policy and new proposals.

> - Identify and train individuals from the community who are interested in contributing to Ansible.

Author: I think `Review things (collections for now) for compliance with the policies.` and `Review collections submitted for inclusion in the Ansible package and decide whether include them or not.` are more similar except that the first one is generic and the second one is specific to collections.

> - Identify and train individuals from the community who are interested in contributing to Ansible.

Reviewer 2: if we have handling of Inclusion Requests as a separate paragraph in the doc (and IMO it's an important role), I believe this should be mentioned in this list (of responsibilitie).

to this `GitHub
discussion <https://github.com/ansible-collections/ansible-inclusion/discussions/new>`__
and requests regarding new policy/proposals should be submitted through
a new pull request to
`ansible-collections/overview <https://github.com/ansible-collections/overview>`__
GitHub repo.

Also, we earlier used a `Community Working Group Meeting
Agenda <https://github.com/ansible/community/issues/539>`__ to track
agenda topics that needed to be discussed in the weekly Ansible
Community IRC meetings. Now, to categorize each topic separately we’ve
created a new GitHub repo
“`community-topics <https://github.com/ansible-community/community-topics>`__”.
If you want to discuss an idea, improvements or submitted new
Policy/Proposals & New Collection Inclusion Requests then create a `new
issue <https://github.com/ansible-community/community-topics/issues/new>`__
in the `repo <https://github.com/ansible-community/community-topics>`__
as a topic, and it’ll be discussed publicly in the weekly IRC meeting.
After every meeting, the meeting minutes/summaries and meeting logs will
be posted to the `original
issue <https://github.com/ansible/community/issues/539>`__.

> If you cannot join the meeting or from another time zone then you can
> directly use GitHub Discussions feature to discuss your new
> proposals/collection inclusion requests and the steering committee will
> discuss the final conclusions during the weekly IRC meeting and then
> will vote on them.
> 

Reviewer 2: ```suggestion
```
As we are moving to the async discussion / voting model, i think we could remove this for now (the async voting has already started happening)

Meetings:
=========

> Ansible Community meeting happens on a weekly basis on
> #ansible-community IRC channel every Wednesday `at 18:00
> UTC <https://raw.githubusercontent.com/ansible/community/main/meetings/ical/community.ics>`__.

Reviewer 2: ```suggestion
See the Community Working Group meeting `schedule <https://github.com/ansible/community/blob/main/meetings/README.md#wednesdays.>`.
```
It's imo better to have this in one place and reference from here (besides possible seasonal changes, it's gonna drastically change soon, i.e. it'll probably get floating to cover more TZs).

Agenda
~~~~~~

> You can find the Ansible Community Meeting agenda at `Community Working
> Group Meeting
> Agenda. <https://github.com/ansible/community/issues/539>`__

Reviewer 2: ```suggestion
You can find the discussion topics in the `community-topics <https://github.com/ansible-community/community-topics>`_ repository.

For topics that are going to be discussed in the next meeting, see `here <https://github.com/ansible-community/community-topics/labels/next_meeting>`_.

For a short summary on the past topics already discussed in the Community Meeting, see the `Meeting Agenda issue <https://github.com/ansible/community/issues/539>`_.
```

Minutes of Meeting
~~~~~~~~~~~~~~~~~~

> Ansible Community weekly meeting minutes can be found at the `fedora
> meetbot
> site <https://meetbot.fedoraproject.org/sresults/?group_id=ansible-community&type=channel>`__
> and the same will be posted to `Ansible Devel Mailing
> List <https://groups.google.com/g/ansible-devel>`__ after every meeting.

Reviewer 2: ```suggestion
Ansible Community weekly meeting minutes can be found at the `fedora meetbot site <https://meetbot.fedoraproject.org/sresults/?group_id=ansible-community&type=channel>`_ and the same will be posted to `Ansible Devel Mailing List <https://groups.google.com/g/ansible-devel>`_ after every meeting.
```

> Mailing List:
> -------------
> 
> Any important announcements regarding Ansible Community Project will be
> posted to ``ansible-devel@googlegroups.com``.

Reviewer 2: ```suggestion
Mailing List:
-------------

Any important announcements regarding Ansible Community Project will be
posted to ``ansible-devel@googlegroups.com``.
```
Is it relevant? If it's relevant for core only, maybe we should exclude the paragraph not to confuse folks. I would put Bullhorn inestead.

Author: Initially, we thought of creating a new mailing list for our own but Toshio suggested to use the existing ansible-devel google group as that group also includes folks from the community.

Reviewer 2: Have it worked so far? If not
1. It should be actively used:) i.e. at least, the decisions made on community topics should be delivered via this
2. Or it should be removed because someone can rely on this source and think that nothing happens

************************************
Ansible Community Steering Committee
************************************

The Steering Committee’s role is to provide continuity, guidance, and
suggestions to the Ansible community, and to ensure delivery of the Ansible
Community package. The Steering Committee should reflect the scope and
breadth of the Ansible Community.

Mission:
========

The Steering Committee helps decide the technical direction of the

> Ansible Upstream project and is also responsible for avowal of new

Reviewer 2: ```suggestion
Ansible project and is also responsible for avowal of new
```

- Review collections submitted for inclusion in the Ansible package and decide whether include them or not.

Members:
========

*Key: Full name (IRC/Matrix Nickname)*

1. Felix Fontein (``felixfontein``)
2. Tadej Borovšak (``tadeboro``)
3. James Cassell (``cyberpear``)
4. John Barker (``gundalow``)
5. Andrew Klychkov (``andersson007_``)
6. Alicia Cozine (``acozine``)
7. Sorin Sbarnea (``zbr``)
8. Jill Rouleau (``jillr``)
9. Brad Thornton (``cidrblock``)
10. Dylan Silva (``thaumos``)

-  Currently, all the Steering Committee members are selected based on
   their active contribution to the Ansible Project.
-  In future, the selection process will be based on the election model
   through community.

New Policy/Proposals & Inclusion Requests:
==========================================

Requests regarding new collection inclusion requests should be submitted
via this `GitHub discussion <https://github.com/ansible-collections/ansible-inclusion/discussions/new>`_.

Requests regarding new inclusion policy/proposals should be submitted through a new pull request to `ansible-collections/overview <https://github.com/ansible-collections/overview>`_ GitHub repository.

> Also, we earlier used a `Community Working Group Meeting Agenda <https://github.com/ansible/community/issues/539>`_ to track
> agenda topics that needed to be discussed in the weekly Ansible Community `IRC/Matrix <https://docs.ansible.com/ansible/devel/community/communication.html#real-time-chat>`_ meetings. Now, to categorize each topic separately we’ve
> created a new GitHub repository `community-topics <https://github.com/ansible-community/community-topics>`_.

Reviewer 2: ```suggestion
For topics that need to be discussed, the Steering Committee uses the `community-topics repository <https://github.com/ansible-community/community-topics>`_. `IRC/Matrix <https://docs.ansible.com/ansible/devel/community/communication.html#real-time-chat>`_ meeting summaries are posted in the `Community Working Group Meeting Agenda <https://github.com/ansible/community/issues/539>`_ issue.
```

If you want to discuss an idea that can impact the community, collection best practices, inclusion policy, submitted new Policy/Proposals & New Collection Inclusion Requests or other topics the steering committee is responsible for, create a new issue in the `community-topics repository <https://github.com/ansible-community/community-topics>`_ as a discussion topic, and it’ll be discussed asynchronously in the issue itself.

Once the filed issue topic is thorough, the Steering Committee will
discuss the final conclusions and then will vote directly on the issue

> Depending on the policy/collection inclusion requests/ideas you propose, as you prepare your proposal you may need to consider requirements pertaining to research involving [Ansible Community Code of Conduct](https://docs.ansible.com/ansible/latest/community/code_of_conduct.html#code-of-conduct), [Ansible Collection Inclusion Checklist](https://github.com/ansible-collections/overview/blob/main/collection_requirements.rst) and [Ansible Community Guide](https://docs.ansible.com/ansible/latest/community/index.html).

Reviewer 2: ```suggestion
Depending on the policy/collection inclusion requests/ideas you propose, as you prepare your proposal you may need to consider requirements pertaining to research involving `Ansible Community Code of Conduct <https://docs.ansible.com/ansible/latest/community/code_of_conduct.html#code-of-conduct>`_, `Ansible Collection Inclusion Checklist <https://github.com/ansible-collections/overview/blob/main/collection_requirements.rst>`_ and `Ansible Community Guide <https://docs.ansible.com/ansible/latest/community/index.html>`_.
```

************************************
Ansible Community Steering Committee
************************************

The Steering Committee’s role is to provide continuity, guidance, and
suggestions to the Ansible community, and to ensure delivery of the Ansible
Community package. The Steering Committee should reflect the scope and
breadth of the Ansible Community.

Mission:
========

The Steering Committee helps decide the technical direction of the

> Ansible Upstream project and is also responsible for avowal of new

Reviewer 2: I would suggest removing `Upstream`. It should be fine as the committee does not decide but `helps` decide.

> Ansible Upstream project and is also responsible for avowal of new

Author: pushed the changes, Andrei! Thank you!

proposals (feature idea, new policy in collection world), new collection
requests, empower contributors, maintainers & reviewers and other
technical aspects regarding inclusion and packaging.

Committee Member Responsibilities:
==================================

The members in the committee handle the below:

- Design policy and procedure for the collection world.
- Review things (collections for now) for compliance with the policies. 

> - Help create our deliverables (community.general, community.network, the ansible-tarball, docs website).

Reviewer 3: ```suggestion
- Help create our deliverables (community.general, community.network, the ``ansible`` tarball, docs website).
```

6. Alicia Cozine (``acozine``)
7. Sorin Sbarnea (``zbr``)
8. Jill Rouleau (``jillr``)
9. Brad Thornton (``cidrblock``)
10. Dylan Silva (``thaumos``)

-  Currently, all the Steering Committee members are selected based on
   their active contribution to the Ansible Project.
-  In future, the selection process will be based on the election model
   through community.

New Policy/Proposals & Inclusion Requests:
==========================================

Requests regarding new collection inclusion requests should be submitted
via this `GitHub discussion <https://github.com/ansible-collections/ansible-inclusion/discussions/new>`_.

Requests regarding new inclusion policy/proposals should be submitted through a new pull request to `ansible-collections/overview <https://github.com/ansible-collections/overview>`_ GitHub repository.

Also, we earlier used a `Community Working Group Meeting Agenda <https://github.com/ansible/community/issues/539>`_ to track
agenda topics that needed to be discussed in the weekly Ansible Community `IRC/Matrix <https://docs.ansible.com/ansible/devel/community/communication.html#real-time-chat>`_ meetings. Now, to categorize each topic separately we’ve

> created a new GitHub repo `community-topics <https://github.com/ansible-community/community-topics>`_.

Reviewer 3: ```suggestion
created a new GitHub repository `community-topics <https://github.com/ansible-community/community-topics>`_.
```

If you want to discuss an idea that can impact the community, collection best practices, inclusion policy, submitted new Policy/Proposals & New Collection Inclusion Requests or other topics the steering committee is responsible for, create a new issue in the `community-topics repository <https://github.com/ansible-community/community-topics>`_ as a discussion topic, and it’ll be discussed publicly in the weekly community `IRC/Matrix <https://docs.ansible.com/ansible/devel/community/communication.html#real-time-chat>`_ meeting and/or asynchronously in the issue itself.

After every meeting, the meeting minutes/summary and meeting logs are posted to the `original issue <https://github.com/ansible/community/issues/539>`_.

If you cannot join the meeting or you are in another time zone then you can
directly use GitHub Discussions feature to discuss your new
proposals/collection inclusion requests and the steering committee will

> discuss the final conclusions during the weekly `IRC/Matrix <https://docs.ansible.com/ansible/devel/community/communication.html#real-time-chat>`_ meeting and then will vote on them.

Reviewer 3: I'm not sure how much sense it makes to document that now since we're in the process of stopping to vote in the meetings and only vote asynchronously. I don't think we plan to have any more votes in the meeting at all.

CC @tadeboro

Reviewer 2: We could remove this, add `FIXME ...`, and elaborate later after the official decision about moving to async discussion / voting is made.

Reviewer 4: I agree with @felixfontein here. We have already effectively stopped voting during meetings (we direct people to vote in the issue), so there is no need to document something that we know is on its deathbed.

Reviewer 2: Also imo it makes sense to mention the labels that we use. We can achieve this with:
1. Listing them here with short explanation (especially `next_meeting` and `active-vote`, thanks @felixfontein for the idea).
2.  Or we could list them in the community-topic's README and mention the labels here pointing to the README for details.
3. Maybe we shouldn't mention them anywhere at all:)

What do you folks think?

Reviewer 3: @Ompragash we might not even discuss it during a meeting. It could happen that an issue in community-topics is discussed completely asynchronously and never mentioned in a meeting.

Reviewer 3: @Andersson007 I think we can actually slowly retire the `next_meeting` label now that we are going async.

Reviewer 2: @felixfontein agreed. Though maybe there will be special cases that we will want to discuss in real time

Reviewer 2: so I wouldn't rush:) or maybe replace it with something else like `meeting_discussion`

Author: @Andersson007 I would also mention the [Project Board](https://github.com/orgs/ansible-community/projects/2/views/5) that Greg has created and we can directly use them instead of labels. It's more effective than labels in terms of categorizing the topics and all we have to do is change the status of the issue topic on the right pane like below:
![Screenshot 2022-01-05 at 6 26 23 PM](https://user-images.githubusercontent.com/21008429/148221140-2b91d424-008a-4ea1-a887-e79469ebc171.png)

Reviewer 2: @Ompragash i would leave labels for now:) they can be helpful for us and for occasional people

7. Sorin Sbarnea (``zbr``)
8. Jill Rouleau (``jillr``)
9. Brad Thornton (``cidrblock``)
10. Dylan Silva (``thaumos``)

-  Currently, all the Steering Committee members are selected based on
   their active contribution to the Ansible Project.
-  In future, the selection process will be based on the election model
   through community.

New Policy/Proposals & Inclusion Requests:
==========================================

Requests regarding new collection inclusion requests should be submitted
via this `GitHub discussion <https://github.com/ansible-collections/ansible-inclusion/discussions/new>`_.

Requests regarding new inclusion policy/proposals should be submitted through a new pull request to `ansible-collections/overview <https://github.com/ansible-collections/overview>`_ GitHub repository.

Also, we earlier used a `Community Working Group Meeting
Agenda <https://github.com/ansible/community/issues/539>`_ to track
agenda topics that needed to be discussed in the weekly Ansible
Community IRC meetings. Now, to categorize each topic separately we’ve
created a new GitHub repo `community-topics <https://github.com/ansible-community/community-topics>`_.

> If you want to discuss an idea that can impact the community, collection best practices, inclusion policy, submitted new Policy/Proposals & New Collection Inclusion Requests or other topics the steering committee is responsible for, create a new issue in the `community-topics repository <https://github.com/ansible-community/community-topics>`_ as a discussion topic, and it’ll be discussed publicly in the weekly community IRC meeting and/or asynchronously in the issue itself.

Reviewer 2: ```suggestion
If you want to discuss an idea that can impact the community, collection best practices, inclusion policy, submitted new Policy/Proposals & New Collection Inclusion Requests or other topics the steering committee is responsible for, create a new issue in the `community-topics repository <https://github.com/ansible-community/community-topics>`_ as a discussion topic, and it’ll be discussed publicly in the weekly community `IRC/Matrix <https://docs.ansible.com/ansible/devel/community/communication.html#real-time-chat>`_ meeting and/or asynchronously in the issue itself.
```
